### PR TITLE
CCW ticket issue #462 changes

### PR DESCRIPTION
**Pull request by orhanf**
LSTM and GatedRecurrent `dim` and `activation` parameter passing harmonized.
For GatedRecurrent default value for `activation` is also set to `Tanh`.
Unit tests changed respectively.

### CODE REVIEW

File: blocks/bricks/recurrent.py [hunk 1]
```diff
@@ -439,9 +439,10 @@ class GatedRecurrent(BaseRecurrent, Initializable):
 
     Parameters
     ----------
-    activation : :class:`.Brick`
-        The brick to apply as activation.
-    gated_activation : :class:`.Brick` or None
+    activation : :class:`.Brick` or None
```
Comment (rizar) [inline blocks/bricks/recurrent.py]: Can you please permute the entries of `Parameters` section in the same order as argument of the `__init__` method?

File: tests/bricks/test_recurrent.py [hunk 2]
```diff
@@ -143,14 +143,17 @@ def sigmoid(x):
...
+        self.gated = GatedRecurrent(dim=3,
+                                    activation=Tanh(),
+                                    gate_activation=Tanh(),
+                                    weights_init=Constant(2))
         self.gated.initialize()
-        self.reset_only = GatedRecurrent(
-            dim=3, weights_init=IsotropicGaussian(),
-            activation=Tanh(), gate_activation=Tanh(),
-            use_update_gate=False, seed=1)
+        self.reset_only = GatedRecurrent(dim=3,
```
Comment (rizar) [inline tests/bricks/test_recurrent.py]: Although technically PEP8 allows that, please do not use this Groundhog style indentation, that is one line per one keyword argument. It takes too much space and we in Blocks sort of don't do it. Sorry for being picky!

**Comment by rizar** [comment]
I have only 2 comments and then we can merge!

File: tests/bricks/test_recurrent.py (hunk 2 above)
Comment (orhanf) [inline tests/bricks/test_recurrent.py]: It's okay, good to know the convention. Fixed it.

**Comment by orhanf** [comment]
fixed according to @rizar comments

**Comment by rizar** [comment]
Very nice, thanks!

**Pull request merged**